packages/utils/src/modules-sdk/abstract-module-service-factory.ts

```diff
@@ -155,17 +171,34 @@ export type AbstractModuleService<
     ): Promise<Record<string, string[]> | void>
   }
 } & {
-  [K in keyof TOtherModelNamesAndAssociatedDTO as `restore${ExtractPluralName<
-    TOtherModelNamesAndAssociatedDTO,
-    K
-  > &
+  [K in keyof ModelsConfig as `restore${ExtractPluralName<ModelsConfig, K> &
     string}`]: {
     <TReturnableLinkableKeys extends string>(
       primaryKeyValues: string | object | string[] | object[],
       config?: RestoreReturn<TReturnableLinkableKeys>,
       sharedContext?: Context
     ): Promise<Record<string, string[]> | void>
   }
+} & {
+  [ModelName in keyof ModelsConfig as CreateMethodName<
+    ModelsConfig,
+    ModelName
+  >]: {
+    (
+      data: ModelsConfig[ModelName]["create_dto"][],
+      sharedContext?: Context
+    ): Promise<ModelsConfig[ModelName]["dto"][]>
+  }
+} & {
+  [ModelName in keyof ModelsConfig as UpdateMethodName<
+    ModelsConfig,
+    ModelName
+  >]: {
+    (
+      data: ModelsConfig[ModelName]["update_dto"][],
+      sharedContext?: Context
+    ): Promise<ModelsConfig[ModelName]["dto"][]>
+  }
 }
```

Comment:
todo:we should add the overload as per our convention

Reply:
Will you add the single data along side the array data?

Reply:
I did already, should be in the new commit